### DISCUSSION TITLE
Is it possible to keep my changes to the card templates while upgrading?

### DISCUSSION THREAD

**Discussion by bdesham** [axelboc/anki-ultimate-geography#400] [Q&A] [answered]
I use Ultimate Geography and I’ve made changes to some of the card layouts to match my “house style.” CrowdAnki offers the option to ignore changes to certain *fields* when you upgrade, but it seems like you just have to accept changes to the card templates. UG doesn’t use that many card types, so I suppose I could just copy my personalized templates, upgrade, and then replace the templates again; is it possible to do this with CrowdAnki and I’m just not seeing the option?

**Comment by aplaice** [comment, marked as answer]
Unfortunately, not easily just with the import dialog. (At least not yet — it's somewhere on the CrowdAnki roadmap.)

For now, as a _possibly_ slightly faster alternative to manually copying the templates, I'd recommend 

1. Exporting your existing deck with CrowdAnki.
2. In that exported json removing the contents of the "notes" field (just leaving `"notes": []) and the "media_files" field (again leaving just `"media_files": []`)
3. Importing 4.1 with CrowdAnki.
4. Importing your edited json.

(This will import the new field contents and media, while keeping your edits to the templates and to the deck options, with the very tiny downside of overwriting the 4.1 description.)

**Reply by bdesham** [reply]
That worked well, thank you!

**Comment by axelboc** [comment]
There's now a section in the `README` on [deck customisation](https://github.com/axelboc/anki-ultimate-geography#changes-that-are-or-can-be-preserved-), which describes an alternative way to keep your changes to the card templates and styles.
